### ISSUE TITLE
Enabler: Replace paratest with run-tests.sh

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#2908]
# Replace paratest (`blt custom:paratest`) with Drupal core run-tests.sh (`blt tests:drupal:run`) 

> [!NOTE]
> I have seen some mentions of run-tests going away in the future. The point behind this issue is that it would be nice to get away from Paratest and implement our tests in a less custom way.

Our custom Paratest command, which was built to run multiple PHPUnit tests concurrently, uses Paratest as the underlying test runner. This runner has some quirks, the most problematic is that it swallows deprecation notices yet still returns a value > 1 as an exit code. (Meaning a failure) Running `blt test:phpunit` to track down the offending code takes no less than 1 hour. (On a fast machine)

With the advent of BLT 10, and some changes with Drupal >8.7, we can now use Drupal's own `run-tests.sh` command. This allows multiple PHPUnit tests to be run concurrently, but (hopefully) does not swallow deprecation notices. 

> **ESTIMATE** NN

### Resources:
* [BLT Docs](https://docs.acquia.com/blt/developer/testing/#drupal-s-run-tests-sh-script)
* [Drupal Docs](https://www.drupal.org/docs/testing/phpunit-in-drupal/running-tests-through-command-line-with-run-testssh)

### Progress / Steps so far...
1. Installed sqlite into the docker container. This is a requirement.
2. Updated the following in blt.yml in the tests section. This took a NUMBER of rounds to get this far.
    ```
    tests:
      phpunit:
        - { config: '${repo.root}/phpunit.xml', directory: '${docroot}/modules/custom' }
        - { config: '${repo.root}/phpunit.xml', directory: '${docroot}/profiles/custom/cgov_site' }
      drupal:
        test-runner: run-tests-script
        sudo-run-tests: false
        simpletest-base-url: 'http://simpletest'
        simpletest-db: 'mysql://drupal:drupal@db/simpletest'
        drupal-tests:
          -
            concurrency: 8
            types:
              - 'PHPUnit-Unit'
              - 'PHPUnit-Kernel'
              - 'PHPUnit-Functional'
            directory: '${docroot}/modules/custom'
            all: false
            url: '${tests.drupal.simpletest-base-url}'
            sqlite: '${tests.drupal.sqlite}'
          -
            concurrency: 8
            types:
              - 'PHPUnit-Unit'
              - 'PHPUnit-Kernel'
              - 'PHPUnit-Functional'
            directory: '${docroot}/profiles/custom/cgov_site'
            all: false
            url: '${tests.drupal.simpletest-base-url}'
            sqlite: '${tests.drupal.sqlite}'
    ```
This results in the tests beginning to run, but additional errors pop in.

### Next Steps
1. Add `@group` annotations to EVERY test class. This should be at least the name of the module. Additionally another annotation for the test WITHIN that module should be added.
    * Also test class namespaces need to be correct. There were some bad namespaces that caused issues.
    * `@coversDefaultClass` and `@covers` annotation must be correct.
    * Basically, ParaTest and PHPUnit was a lot more lax with bad code/annotations. 
3. ~~Find a better place, and configure it, for the sqllite db instead of the docroot.~~    * The DB was being created at docroot because of a wrong `sqlite` parameter. Using the above cofig has no issue. 
2. Update the Docker container to have sqlite install upon creation
3. Figure out what needs to be done for running this in pipelines.
4. Document how to run all tests and how to run just a single test class.
5. Keep phpunit working so we can debug tests.
6. Add `{repo_root}/tmp` to gitignore.

**Comment by bryanpizzillo** [comment]
Ok, with PHP 7.3 we need to update to PHPUnit 7.

**Comment by whitakerwh** [comment]
@bryanpizzillo to look into run-tests to determine if this ticket still needs to be done.

**Comment by bryanpizzillo** [comment]
I ran some local tests with `run-tests.sh` to figure out concurrency. These tests were run in `docroot/modules`.
* Paratest
   *  4 Threads - 5m41s
* Run-tests.sh 
   * 16 concurrency - 4m21s
   * 8 concurrency - 4m29s
   * 4 concurrency - 5m12s

GitHub may require different numbers.

**Comment by bryanpizzillo** [comment]
Prototype Branch: https://github.com/NCIOCPL/cgov-digital-platform/compare/develop...prototype/run-tests

**Comment by bryanpizzillo** [comment]
There may be some changes needed for D11 once D11 it is installed. But this is not blocking the tests, but rather the old `@group` annotations are deprecated and the newer php attributes need to be used. However, it looks like 11.2 is when they added the attributes, so it is not something that can be added earlier. I note this because there may be deprecation notices raised when running tests in D11.2 - hopefully not since the deprecated code should be in the runners.

https://www.drupal.org/project/drupal/issues/3497431
